test/e2e/config/vsphere-ci.yaml

```diff
@@ -6,165 +6,200 @@
-    versions:
-      - name: v0.7.12
-        value: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/releases/download/v0.7.12/infrastructure-components.yaml
-        type: url
-        contract: v1alpha3
-        replacements:
-          - old: "imagePullPolicy: Always"
-            new: "imagePullPolicy: IfNotPresent"
-        files:
-          # TODO: v1a3 cluster-template includes WORKLOAD_CONTROL_PLANE_ENDPOINT_IP
-          - sourcePath: "../data/metadata.yaml"
-          - sourcePath: "../data/infrastructure-vsphere/capi-upgrades/v1alpha3/cluster-template.yaml"
-      - name: v0.8.2
-        value: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/releases/download/v0.8.2/infrastructure-components.yaml
-        type: url
-        contract: v1alpha4
-        replacements:
-          - old: "imagePullPolicy: Always"
-            new: "imagePullPolicy: IfNotPresent"
-        files:
-          - sourcePath: "../data/metadata.yaml"
-          - sourcePath: "../data/infrastructure-vsphere/capi-upgrades/v1alpha4/cluster-template.yaml"
-      - name: v1.0.1
-        value: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/releases/download/v1.0.1/infrastructure-components.yaml
-        type: url
-        contract: v1beta1
-        replacements:
-          - old: "imagePullPolicy: Always"
-            new: "imagePullPolicy: IfNotPresent"
-        files:
-          # Add a cluster template
-          - sourcePath: "../data/infrastructure-vsphere/cluster-template.yaml"
-          - sourcePath: "../data/infrastructure-vsphere/cluster-template-remote-management.yaml"
-          - sourcePath: "../data/metadata.yaml"
+- name: kubeadm
+  type: ControlPlaneProvider
+  versions:
+  - name: v1.6.1
+    # Use manifest from source files
+    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.6.1/control-plane-components.yaml"
+    type: "url"
+    contract: v1beta1
+    files:
+    - sourcePath: "../data/shared/main/v1beta1/metadata.yaml"
+    replacements:
+    - old: "imagePullPolicy: Always"
+      new: "imagePullPolicy: IfNotPresent"
+  - name: v1.5.4
+    # Use manifest from source files
+    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.5.4/control-plane-components.yaml"
+    type: "url"
+    contract: v1beta1
+    files:
+    - sourcePath: "../data/shared/v1.8/v1beta1/metadata.yaml"
+    replacements:
+    - old: "imagePullPolicy: Always"
+      new: "imagePullPolicy: IfNotPresent"
+  - name: v1.4.9
+    # Use manifest from source files
+    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.4.9/control-plane-components.yaml"
+    type: "url"
+    contract: v1beta1
+    files:
+    - sourcePath: "../data/shared/v1.7/v1beta1/metadata.yaml"
+    replacements:
+    - old: "imagePullPolicy: Always"
+      new: "imagePullPolicy: IfNotPresent"
+
+- name: vsphere
+  type: InfrastructureProvider
+  versions:
+  - name: v1.9.0
```

Comment:
@sbueringer @chrischdi Could you take a look at section for v1.9.0? I copied from v1.9.99 and made some changes.

Reply:
I'll take a closer look once we merged the CAPV PR (should only take a few days)

Reply:
Thank you. Will the CAPV PR be merged to release-1.9 branch?

Reply:
No. That one is specifically for main

Reply:
Do you suggest CPI to clone main branch of CAPV for e2e testing?  Currently CPI clones release-1.0 branch and I changed it to release-1.9 branch in this PR.

Reply:
I have to take a closer look

Reply:
Took a look and discussed it with Christian. Let's try to write a relatively straightforward e2e config that uses the version from the release-1.9 CAPV branch.

On a high-level
* We want to use CAPI v1.6.1 & CAPV v1.9
* We want to keep the config simple, so it's simple to upgrade later
* If we are aware of any changes needed in CAPI or CAPV upstream for Kubernetes 1.30 we'll let you know so you can pick up CAPI / CAPV beta / RC versions 


Based on that:
* We only have to configure one version of CAPI / CAPV providers (CAPV has more as it also tests CAPI upgrades)

Reply:
The providers config should then look like this 

```yaml
providers:

- name: cluster-api
  type: CoreProvider
  versions:
  - name: v1.6.1
    # Use manifest from source files
    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.6.1/core-components.yaml"
    type: "url"
    contract: v1beta1
    files:
    - sourcePath: "../data/shared/main/v1beta1/metadata.yaml"
    replacements:
    - old: "imagePullPolicy: Always"
      new: "imagePullPolicy: IfNotPresent"

- name: kubeadm
  type: BootstrapProvider
  versions:
  - name: v1.6.1
    # Use manifest from source files
    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.6.1/bootstrap-components.yaml"
    type: "url"
    contract: v1beta1
    files:
    - sourcePath: "../data/shared/main/v1beta1/metadata.yaml"
    replacements:
    - old: "imagePullPolicy: Always"
      new: "imagePullPolicy: IfNotPresent"

- name: kubeadm
  type: ControlPlaneProvider
  versions:
  - name: v1.6.1
    # Use manifest from source files
    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.6.1/control-plane-components.yaml"
    type: "url"
    contract: v1beta1
    files:
    - sourcePath: "../data/shared/main/v1beta1/metadata.yaml"
    replacements:
    - old: "imagePullPolicy: Always"
      new: "imagePullPolicy: IfNotPresent"

- name: vsphere
  type: InfrastructureProvider
  versions:
  - name: v1.9.0
    # Use manifest from source files
    value: "https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/releases/download/v1.9.0/infrastructure-components.yaml"
    type: "url"
    contract: v1beta1
    files:
    # Add a cluster template
    - sourcePath: "../../../test/e2e/data/infrastructure-vsphere/main/cluster-template.yaml"
    - sourcePath: "../data/shared/main/v1beta1_provider/metadata.yaml"

```

(should cover all the suggested changes here)